d2core/d2map/d2mapentity/factory.go

```diff
@@ -102,6 +102,7 @@ func (f *MapEntityFactory) NewPlayer(id, name string, x, y, direction int, heroT
 		isInTown:     true,
 		isRunning:    false,
 		Gold:         gold,
+		Act:          1,
```

Comment:
This would set new players to start in `DifficultyNightmare`

Reply:
Does it? It only sets the Act, which is an attribute of the player. The difficulty level is not an attribute of the player, it gets set nowhere at the moment and is only used in the save player function where it it hardcoded to d2enum.DifficultyNormal.